### ISSUE TITLE
Improving reproducibility in science by adding provenance tracking to the EcoData Retriever 

### ISSUE THREAD

**Issue by prabh27** [weecology/retriever#454]
I had a look at the provenance library. Let us discuss the requirements so that we can get started on the design of the problem, and see if this is a good fit or not. 
Link -> http://archive.is/74MON
@ethanwhite

**Comment by henrykironde** [comment]
@prabh27 sounds good, we could discus this next week. Currently at a workshop.

**Comment by mbjones** [comment]
@ethanwhite and @prabh27 That would be awesome if you started tracking provenance!

Can I suggest you also consider the DataONE work on provenance tracking for ecological and environmental data and analyses?  We have developed an extension to PROV for modeling the science workflow of derived data products -- see [ProvONE](https://purl.dataone.org/provone-v1-dev) as well as provenance tracking tools in both R (see [recordr](https://github.com/NCEAS/recordr)) and matlab (see [matlab-dataone-toolbox](https://github.com/DataONEorg/matlab-dataone)).  If you start generating provenance traces from the EcoData Retriever, it would be nice if that provenance metadata were compatible with the formats we are publishing in the DataONE federation of data repositories.

I gave an overview talk on our tools and model at ESA and AGU last year which may be of interest. 

> Jones, Matthew; Jones, Christopher; Walker, Lauren; Slaughter, Peter; Leinfelder, Benjamin (2015): Reproducible science via semantics and provenance for ecological data. figshare. https://dx.doi.org/10.6084/m9.figshare.1512516.v1
